builder/dockerfile/dispatchers_windows.go

```diff
@@ -17,8 +17,27 @@ func normaliseWorkdir(current string, requested string) (string, error) {
 		return "", fmt.Errorf("cannot normalise nothing")
 	}
 
-	current = filepath.FromSlash(current)
-	requested = filepath.FromSlash(requested)
+	// `filepath.Clean` will replace "" with "." so skip in that case
+	if current != "" {
+		current = filepath.Clean(current)
+	}
+	if requested != "" {
+		requested = filepath.Clean(requested)
+	}
+
+	// If either current or requested in Windows is:
+	// C:
+	// C:.
+	// then an error will be thrown as the definition for the above
+	// refers to `current directory on driver C:`
+	// Since filepath.Clean() will automatically normalize the above
+	// to `C:.`, we only need to check the last format
+	if len(current) == 3 && strings.HasSuffix(current, ":.") {
+		return "", fmt.Errorf("cannot use indefinitive current directory '%s'", current)
```

Comment:
`"%s is not a directory. If you are specifying a driver letter, please add a trailing '\'"`